src/core/matchers/matchersUtil.js

```diff
@@ -167,11 +167,19 @@ getJasmineRequireObj().matchersUtil = function(j$) {
 
     if (result) {
       // Objects with different constructors are not equivalent, but `Object`s
-      // from different frames are.
-      var aCtor = a.constructor, bCtor = b.constructor;
-      if (aCtor !== bCtor && !(isFunction(aCtor) && (aCtor instanceof aCtor) &&
-        isFunction(bCtor) && (bCtor instanceof bCtor))) {
-        return false;
+      // or `Array`s from different frames are.
+      var areArrays = className === '[object Array]';
+      if (!areArrays) {
+        if (typeof a != 'object' || typeof b != 'object') {
```

Comment:
This line appears to be a duplicate of [line 126](https://github.com/jasmine/jasmine/pull/767/files#diff-4b1cbd917e62d27b2645d46dd0d01be3L126)